### DISCUSSION TITLE
goBack will exit APP on iOS

### DISCUSSION THREAD

**Discussion by PeterCheungHK** [react-navigation/react-navigation#11574] [Q&A]
I just use will simple code 
`import React from 'react';
import { View, Text, StyleSheet, Button } from 'react-native';

function AddMemorial({ navigation }) {
  return (
    <View style={styles.container}>
      <Text>Profile Screen</Text>
      <Text>Friends: </Text>
      <Button title="Go back" onPress={() => navigation.goBack(null)} />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default AddMemorial;
`

in android the button can retune to the home page , but the iOS will exit the APP

**Comment by PeterCheungHK** [comment]
It should be this code
```
import React from 'react';
import { View, Text, StyleSheet, Button } from 'react-native';

function AddMemorial({ navigation }) {
  return (
    <View style={styles.container}>
      <Text>Profile Screen</Text>
      <Text>Friends: </Text>
      <Button title="Go back" onPress={() => navigation.goBack(null)} />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
});

export default AddMemorial;
```

**Reply by SammyCodee** [reply]
Facing the same issue as you.

In react-navigation v6, I was using:

_import { createNativeStackNavigator } from "@react-navigation/native-stack";_ ❌

I solved the issue by:

_import { createStackNavigator } from "@react-navigation/stack";_ ✅

_const RootStack = createStackNavigator<RootStackParamList>();
const HomeStack = createStackNavigator<HomeScreenStackingList>();_
